include/coap/coap_dtls.h

```diff
@@ -51,20 +51,51 @@ int coap_dtls_get_log_level(void);
 struct coap_dtls_pki_t;
 
 /**
- * Security setup handler that is used as call-back in coap_context_set_pki()
+ * Security setup handler that is used as full call-back in
+ * coap_context_set_pki().
  * Typically, this will be calling additonal functions like
  * SSL_CTX_set_tlsext_servername_callback() etc.
  *
  * @param context The security context definition - e.g. SSL_CTX * for OpenSSL. 
- *              This will be dependent on the underlying TLS library
- *              - see coap_get_tls_library_version()
+ *                This will be dependent on the underlying TLS library
+ *                - see coap_get_tls_library_version()
+ * @param session The security session definition - e.g. SSL * for OpenSSL. 
+ *                NULL if server call-back.
+ *                This will be dependent on the underlying TLS library
+ *                - see coap_get_tls_library_version()
  * @param setup_data A structure containing setup data originally passed into
- *                  coap_context_set_pki() or coap_new_client_session_pki().
+ *                   coap_context_set_pki() or coap_new_client_session_pki().
  * @return 1 if successful, else 0
  */
-typedef int (*coap_dtls_security_setup_t)(void *context,
+typedef int (*coap_dtls_security_setup_t)(void *context, void* session,
                                         struct coap_dtls_pki_t *setup_data);
 
+/**
+ * CN Validation call-back that can be set up by coap_context_set_pki().
+ * Invoked when libcoap has done the validation checks at the TLS level,
+ * but the application needs to check that the CN is allowed.
+ * CN is the SubjectAltName in the cert, if not present, then the leftmost
+ * Common Name (CN) component of the subject name
+ *
+ * @param cn  The determined CN from the certificate
+ * @param asn1_public_cert  The ASN.1 encoded (DER) X.509 certificate
+ * @param asn1_length  The ASN.1 length
+ * @param session  The coap session associated with the certificate update
+ * @param depth  Depth in cert chain.  If 0, then client cert, else a CA
```

Comment:
depth is derived from X509_STORE_CTX_get_error_depth() which returns an int, so I would prefer this not to be changed, even though the documentation currently states
````
X509_STORE_CTX_get_error_depth() returns a non-negative error depth
````

Reply:
Well, the API in `coap_dtls.h` is agnostic of the TLS library, therefore I do not think that sticking to OpenSSL idiosyncrasies is preferable to cleaning up the API where possible. You would need to repeat the OpenSSL comment here and in the following explain to people repeatedly why this will never be a negative value...

Reply:
Updated to be unsigned as depth should be 0 or greater.